### PR TITLE
feat: support json format output in pnpm pack

### LINKED ISSUE #8764: `pnpm pack` JSON output option
### Contribution

- [ ] I'd be willing to implement this feature ([contributing guide](https://github.com/pnpm/pnpm/blob/main/CONTRIBUTING.md))

### Describe the user story

Allow easy parsing of `pnpm pack` output, to avoid code like this:

- https://github.com/pnpm/pnpm/pull/8707#issuecomment-2478284044

### Describe the solution you'd like

Maybe `pnpm pack --format json`? (like `pnpm outdated --format json` https://github.com/pnpm/pnpm/pull/5582 )

```bash
$ pnpm pack --format json
{
  "tarball": "upleveled-preflight-8.0.4.tgz",
  "contents": [
    "bin/preflight.sh",
    "LICENSE",
    "package.json",
    "README.md",
    "src/checks/allChangesCommittedToGit.ts",
    "src/checks/eslint.ts",
    "src/checks/eslintConfigIsValid.ts",
    "src/checks/linkOnGithubAbout.ts",
    "src/checks/nodeModulesIgnoredFromGit.ts",
    "src/checks/noDependencyProblems/noDependenciesWithoutTypes.ts",
    "src/checks/noDependencyProblems/noUnusedDependencies.ts",
    "src/checks/noExtraneousFilesCommittedToGit.ts",
    "src/checks/noSecretsCommittedToGit.ts",
    "src/checks/preflightIsLatestVersion.ts",
    "src/checks/prettier.ts",
    "src/checks/projectFolderNameMatchesCorrectFormat.ts",
    "src/checks/stylelint.ts",
    "src/checks/stylelintConfigIsValid.ts",
    "src/checks/useSinglePackageManager.ts",
    "src/index.ts",
    "src/util/commandExample.ts",
    "src/util/crossPlatform.ts",
    "src/util/drone.ts",
    "src/util/packageJson.ts",
    "src/util/preflightBinPath.ts",
    "src/util/randomUserAgent.ts"
  ]
}
```

### Describe the drawbacks of your solution

More code to maintain

### Describe alternatives you've considered

My verbose parsing solution

- https://github.com/pnpm/pnpm/pull/8707#issuecomment-2478284044

### PR DESCRIPTION
**Pull request by lvqq**
close #8764

### CODE REVIEW

**Review by zkochan — CHANGES_REQUESTED** [review]
We already use a `--json` flag in other commands (e.g., publish, audit, list, why). So, we should do that here as well instead of `--format=json`

File: releasing/plugin-commands-publishing/src/pack.ts [hunk 1]
```diff
@@ -145,8 +151,15 @@ export async function handler (
   } else {
     packedTarballPath = path.relative(opts.dir, path.join(dir, tarballName))
   }
+  const packedContents = files.sort((a, b) => a.localeCompare(b, 'en'))
+  if (opts.format === 'json') {
+    return JSON.stringify({
+      tarball: packedTarballPath,
+      contents: packedContents,
+    }, null, 2)
```
Comment (zkochan) [inline releasing/plugin-commands-publishing/src/pack.ts]: check what `npm pack --json` prints and maybe do something similar. Probably printing `filename` and `files` similar to npm's output should be enough.

Comment (lvqq) [reply releasing/plugin-commands-publishing/src/pack.ts]: Yeah, that's how `npm` prints. I have updated it

File: .changeset/spicy-needles-peel.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+"@pnpm/plugin-commands-publishing": minor
+---
+
+feat: support `pnpm pack --json` to print packed tarball and contents in json format
```
Comment (zkochan) [inline .changeset/spicy-needles-peel.md]:
```suggestion
Added support for `pnpm pack --json` to print packed tarball and contents in JSON format [#8765](https://github.com/pnpm/pnpm/pull/8765).
```

**Review by zkochan — APPROVED** [review]

**Pull request merged**